Sds trustworthy jwt doc (#4701)
istio/istio.io: content/en/blog/2019/trustworthy-jwt-sds/index.md

---

In Istio 1.3, we are taking advantage of improvements in Kubernetes to issue certificates for workload instances more securely.

When a Citadel Agent (the agent that runs on a node and is responsible for obtaining certificates)
sends a certificate signing request to Citadel to get a certificate for a workload instance,
it includes the JWT that the Kubernetes API server issued representing the service account of the workload instance.
If Citadel can authenticate the JWT, then it extracts the service account name needed to issue the certificate for the workload instance.

The API server prior to Kubernetes 1.12 issues JWTs that present the following problems:

1. The tokens don't have important fields to limit their scope of usage, such as `aud` or `exp`. See [Bound Service Tokens](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/auth/bound-service-account-tokens.md) for more info.
1. The tokens are mounted onto all the pods without a way to opt-out. See [Service Account Token Volumes](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/svcacct-token-volume-source.md) for motivation.

> The `trustworthy` JWTs were introduced in Kubernetes 1.12 to solve the aforementioned issues.

Reviewer: ```suggestion
Kubernetes 1.12 introduces `trustworthy` JWTs to solve these issues.
```

> However, Kubernetes support for the `aud` field to be different than the API server audience became available until [Kubernetes 1.13](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md).

Reviewer 2: ```suggestion
However, Kubernetes' support for the `aud` field to be different than the API server audience became available until [Kubernetes 1.13](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md).
```

Reviewer: There's no need to have Kubenetes twice in the sentence also fixing minor verb tense issue. 
```suggestion
However, support for the `aud` field to have a different value than the API server audience didn't become available with [Kubernetes 1.13](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md).
```

Author: @rcaballeromx The language is messing with me here. So the information that I'm trying to communicate here is k8s didn't support different audience until version 1.3. Is what you're suggesting saying that? I think I missed the `didn't` part in my current PR.

Reviewer: Sorry wrong preposition crept in. Is this better:
```suggestion
However, support for the `aud` field to have a different value than the API server audience didn't become available until [Kubernetes 1.13](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md).
```

> However, Kubernetes didn't support the `aud` to be anything other than the API server audience until [Kubernetes 1.13](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md).

Reviewer: ```suggestion
However, Kubernetes support for the `aud` field to be different from the API server audience became available until [Kubernetes 1.13](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md).
```

> To better secure the mesh, Istio 1.3 only supports `trustworthy` JWTs and requires a specific value for the `aud` field when you enable SDS.

Reviewer 2: ```suggestion
To better secure the mesh, Istio 1.3 only supports `trustworthy` JWTs and requires the specific value `istio-ca` for the `aud` field when you enable SDS.
```

Reviewer: ```suggestion
To better secure the mesh, Istio 1.3 only supports `trustworthy` JWTs and requires the value of the `aud` field to be `istio-ca` when you enable SDS.
```

> For example, if the Certificate Authority is Citadel, Istio will set the `aud` to be `istio-ca` by default.

Reviewer 2: ```suggestion
```

> To better secure the mesh, Istio 1.3 only supports `trustworthy` JWTs and sets the `aud` to a specific value (for example, the value is `istio-ca` in Citadel) when you enable SDS.

Reviewer: ```suggestion
To better secure the mesh, Istio 1.3 only supports `trustworthy` JWTs and requires a specific value for the `aud` field. 
For example, you must set the value to `istio-ca` in Citadel to enable SDS.
```

Before upgrading your Istio deployment to 1.3 with SDS enabled, verify that you use Kubernetes 1.13 or later.

> Make the following considerations based on your platform of choice:

Reviewer: Replace the content from line 26-31 with the following suggestion:
```suggestion
See the [platform-specific setup](/docs/setup/platform-setup/) for any considerations specific to your platform.
```

---
title: Istio 1.3 with SDS Announcement
description: Istio 1.3 with SDS Announcement.
publishdate: 2019-08-23
attribution: Phillip Quy Le (Google)
keywords: [security, PKI, certificate, nodeagent, sds]
---

In Istio 1.3, we have made a change to take advantage of improvements in Kubernetes in order to more
securely issue certificates for workloads.

> When a Citadel Agent (the agent that runs on a node and is responsible for obtaining certificates)

Reviewer: @pitlv2109 I'm fine with us doing a separate security-specific glossary PR.

> When a Citadel Agent (the agent that runs on a node and is responsible for obtaining certificates)

Author: Sounds good.

> sends a Certificate Signing Request to Citadel to get a certificate for a workload,

Reviewer: I couldn't find `Certificate Signing Request` in that medium post. I would suggest we don't even capitalize SDS. The convention is not to capitalize the words of the abbreviation. See https://en.wikipedia.org/wiki/Certificate_signing_request
Be wary of using capitalization for emphasis, that should be avoided because readers will be confused by it.

> sends a Certificate Signing Request to Citadel to get a certificate for a workload,

Author: Got it. Thanks!
